server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestClusterAllocationExplainAction.java

```diff
@@ -47,23 +47,72 @@ public boolean allowSystemIndexAccessByDefault() {
         return true;
     }
 
+    /*
+        The Cluster Allocation Explain API supports both path parameters and parameters passed through the request body, but not both.
+        The API also supports empty requests, which translates to "explain the first unassigned shard you find"
+     */
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
-        final var req = new ClusterAllocationExplainRequest(RestUtils.getMasterNodeTimeout(request));
+        final var clusterAllocationExplainRequest = new ClusterAllocationExplainRequest(RestUtils.getMasterNodeTimeout(request));
+
+        // A request body was passed
         if (request.hasContentOrSourceParam()) {
+            // Supplying parameters alongside a request body is unsupported
+            if (hasAnyParameterBeenPassed(request)) {
+                throw new IllegalArgumentException("Parameters cannot be passed in both the URL and the request body");
+            }
```

Comment:
Is this needed? I would expect that if the request contains a parameter which isn't consumed in this method then that'll automatically throw an IAE - see `org.elasticsearch.rest.BaseRestHandler#handleRequest`, particularly `unconsumedParams`.